RocketApp/app/src/main/java/com/example/rocketapp/LogInActivity.java

```diff
@@ -39,7 +40,12 @@ protected void onCreate(Bundle savedInstanceState) {
         loginBtn.setOnClickListener(v -> {
             EditText usernameEditText = findViewById(R.id.usernameEditText);
             String username = usernameEditText.getText().toString();
```

Comment:
could also do
if (Validate.lengthInRange(userNameEditText, 3, 30, true))
{
     loginOrCreateUser(userNameEditText.getText().toString());
}
and that would cover it and put the error message right below the input